pkg/acsengine/engine.go

```diff
@@ -758,6 +758,44 @@ func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) map[str
 		"HasWindowsSecrets": func() bool {
 			return cs.Properties.WindowsProfile.HasSecrets()
 		},
+		"PopulateClassicModeDefaultValue": func(attr string) string {
+			var val string
+			if !t.ClassicMode {
+				val = ""
```

Comment:
my bad I had assumed we had done it differently. and made it required. because without it being set the deployment will fail.